service-discovery/stork-custom/src/test/java/io/quarkus/ts/stork/custom/StorkCustomLoadBalancerIT.java

```diff
@@ -0,0 +1,86 @@
+package io.quarkus.ts.stork.custom;
+
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+import io.restassured.response.ValidatableResponse;
+
+import junit.framework.AssertionFailedError;
+
+@QuarkusScenario
+public class StorkCustomLoadBalancerIT {
+
+    public static final String PREFIX = "ping-";
+    private static final String DEFAULT_PONG_REPLICA_RESPONSE = "pongReplica";
+    private static final String DEFAULT_PONG_RESPONSE = "pong";
+    private static final String PONG_PORT = getAvailablePort();
+    private static final String PONG_REPLICA_PORT = getAvailablePort();
+
+    @QuarkusApplication(classes = PongResource.class)
+    static RestService pongService = new RestService().withProperty("quarkus.http.port", PONG_PORT);
+
+    @QuarkusApplication(classes = PongReplicaResource.class)
+    static RestService pongReplicaService = new RestService().withProperty("quarkus.http.port", PONG_REPLICA_PORT);
+
+    @QuarkusApplication
+    static RestService mainPingService = new RestService()
+            .withProperty("stork.pong.load-balancer", "simple")
+            .withProperty("stork.pong.service-discovery", "simple")
+            .withProperty("stork.pong-replica.load-balancer", "simple")
+            .withProperty("stork.pong-replica.service-discovery", "simple")
+            .withProperty("stork.pong.service-discovery.pongServicePort", () -> "" + pongService.getPort())
+            .withProperty("stork.pong-replica.service-discovery.pongReplicaServicePort",
+                    () -> "" + pongReplicaService.getPort());
+
+    @Test
+    public void storkCustomLoadBalancer() {
+        Map<String, Integer> uniqueResp = new HashMap<>();
+        final int requestAmount = 100;
+        final int randomLbError = (requestAmount / 2) - 20;
```

Comment:
The `Error` in the name is a little bit confusing. It is, in fact, something like minimum hits, right? Maybe `randomLbMinHits` would be better. WDYT?

Also, just out of curiosity, is the value an estimate, or is it based on something solid?

Reply:
Name changes to the suggested one. 
`> just out of curiosity, is the value an estimate, or is it based on something solid?`

No is an estimation, I mean... a random generator should follow a uniform distribution

```
The probabilities for uniform distribution function are simple to calculate due to the simplicity of the function form. Therefore, there are various applications that this distribution can be used for as shown below: hypothesis testing situations, random sampling cases ...
```
So in theory a `java.util.Random` should distribute the load in a "uniform way", so if we have two instances, as is the case, we should have 50 req per node. However, after some test I saw an error of...10%, to us is not relevant... this is just a fake LB that the only thing that is probing is that the custom LB implementation is working. 

The standard round-robin implementation works [perfectly](https://github.com/quarkus-qe/quarkus-test-suite/blob/main/service-discovery/stork/src/test/java/io/quarkus/ts/stork/StorkLoadBalancerIT.java#L72)!

Reply:
Cool, thanks for the clarification.